src/pkgs.jl

```diff
@@ -58,7 +58,9 @@ function pkg_fileinfo(id::PkgId)
     uuid, name = id.uuid, id.name
     # Try to find the matching cache file
     paths = Base.find_all_in_cache_path(id)
+    sourcepath = Base.locate_package(id)
     for path in paths
+        Base.stale_cachefile(sourcepath, path) === true && continue
```

Comment:
`stale_cachefile` also, of course, parses the cache header internally but perhaps cheap enough to not worry?

Reply:
On my machine it's of order 1ms. I'm not worried about it, thanks for doing this.